src/snowflake/snowpark/_internal/data_source/dbms_dialects/base_dialect.py

```diff
@@ -6,9 +6,29 @@
 from snowflake.snowpark.types import StructType
 
 
+QUERY_TEMPLATE = "SELECT {cols} FROM {table_or_query} {query_input_alias}"
+
+
 class BaseDialect:
     @staticmethod
     def generate_select_query(
-        table_or_query: str, schema: StructType, raw_schema: List[tuple], is_query: bool
+        table_or_query: str,
+        schema: StructType,
+        raw_schema: List[tuple],
+        is_query: bool,
+        query_input_alias: str,
     ) -> str:
-        return f"SELECT * FROM {table_or_query}"
+        cols = []
+        for raw_field in raw_schema:
+            if is_query:
+                cols.append(
+                    f"""{query_input_alias}.`{raw_field[0]}` AS {raw_field[0]}"""
+                )
+            else:
+                cols.append(raw_field[0])
+
+        return QUERY_TEMPLATE.format(
+            cols=", ".join(cols),
+            table_or_query=f"({table_or_query})" if is_query else table_or_query,
+            query_input_alias=query_input_alias if is_query else "",
+        ).strip()
```

Comment:
I see this part of code is repeated multiple times in the subclasses, can we make it another static method on the BaseDialect so that subclasses can reuse